src/parsers/gobblemode.ts

```diff
@@ -27,18 +30,29 @@ function reset() {
 }
 
 /** Receive keypress. If applicable, execute a command. */
-export function parser(keys: KeyEventLike[]) {
+export function parser(keys: MinimalKey[]) {
```

Comment:
wouldn't this PR be much shorter if we wrote/used a KeyEventLikeToMinimalKey function here? Do we gain a lot by changing it everywhere?

Reply:
Why use KeyEventLikeToMinimalKey everywhere when we can use it once or twice in the top of a pipeline and work with internal type everywhere else? It eliminates checking for InstanceOf and allows direct usage of MinimalKey's methods and helper functions

Reply:
Also, why it only treats the first key? Is it guaranteed it always gets no more than one?

Reply:
> Also, why it only treats the first key? Is it guaranteed it always gets no more than one?

Yeah, it only ever gets one key.

controller_content works like this:

```
1. initialise an empty list of key events
2. receive a key event, add it to that list
3. send the list of key events to a parser and get a new list of key events back - go to 1. Or get an ex string to execute - go to 2.
```

gobble mode only ever returns and empty list of key events https://github.com/tridactyl/tridactyl/blob/74d357e1e2bdbc31ce87d06bfc5ac2ec3843e888/src/parsers/gobblemode.ts#L57 so you can only ever get one key.

Parsing multiple keys at once is so that e.g. the normal mode parser can say "yes the g key matches loads of binds, give me it again with the next key and we'll see if it matches a bind".